src/puppetlabs/trapperkeeper/services/status/check.clj

```diff
@@ -0,0 +1,28 @@
+(ns puppetlabs.trapperkeeper.services.status.check
+  "Shared status check functions."
+  (:require [clojure.java.jdbc :as sql])
+  (:import [java.io File IOException]))
+
+(defn db-up?
+  [db-spec]
+  (try (let [select-42 "SELECT (a - b) AS answer FROM (VALUES ((7 * 7), 7)) AS x(a, b)"
+             [{:keys [answer]}] (sql/query db-spec [select-42])]
+         (= answer 42))
+    (catch Exception _
+      false)))
+
+(defn disk-writable?
+  []
```

Comment:
Those are good points, especially about the drawbacks of only using the temporary directory.